.circleci/config.yml

```diff
@@ -0,0 +1,187 @@
+version: 2.1
+
+orbs:
+  slack: circleci/slack@3.4.2
+
+executors:
+  go:
+    docker:
+      - image: circleci/golang:1.12
+    environment:
+      CONSUL_VERSION: 1.7.2
+      GOMAXPROCS: 4
+      GO111MODULE: "on"
+      GOPROXY: https://proxy.golang.org/
+      TEST_RESULTS_DIR: &TEST_RESULTS_DIR /tmp/test-results
+      ARTIFACTS_DIR: &ARTIFACTS_DIR /tmp/artifacts
+      GOFLAGS: "-mod=vendor"
+
+jobs:
+  go-checks:
+    executor:
+      name: go
+    steps:
+      - checkout
+      - run: go mod verify
+      - run: make fmtcheck generate
+      - run:
+          name: verify no code was generated
+          command: |
+            if [[ -z $(git status --porcelain) ]]; then
+              echo "Git directory is clean."
+            else
+              echo "Git is dirty. Run `make fmtcheck` and `make generate` locally and commit any formatting fixes or generated code."
+              git status --porcelain
+              exit 1
+            fi
+
+  go-test:
+    executor:
+      name: go
+    parallelism: 4
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - run:
+          name: install consul
+          command: |
+            curl -sLo consul.zip https://releases.hashicorp.com/consul/${CONSUL_VERSION}/consul_${CONSUL_VERSION}_linux_amd64.zip
+            unzip consul.zip
+            mkdir -p ~/bin
+            mv consul ~/bin
+            echo 'export PATH="~/bin:$PATH"'
+      - run: mkdir -p $TEST_RESULTS_DIR
+      - run:
+          name: Run Go Tests
+          command: |
+            PACKAGE_NAMES=$(go list ./... | circleci tests split --split-by=timings --timings-type=classname)
+            echo "Running $(echo $PACKAGE_NAMES | wc -w) packages"
+            echo $PACKAGE_NAMES
+            gotestsum --format=short-verbose --junitfile $TEST_RESULTS_DIR/gotestsum-report.xml -- -p 2 -cover -coverprofile=cov_$CIRCLE_NODE_INDEX.part $PACKAGE_NAMES
+
+      # save coverage report parts
+      - persist_to_workspace:
+          root: .
+          paths:
+            - cov_*.part
+
+      - store_test_results:
+          path: *TEST_RESULTS_DIR
+      - store_artifacts:
+          path: *TEST_RESULTS_DIR
+
+      - slack/status:
+          fail_only: true
+          only_for_branches: master
```

Comment:
Should we add v0.12 here too?

Reply:
I didn't see any real value in getting a notification for a branch we're not actively developing, but I have no objections to adding it.